### DISCUSSION TITLE
C axis

### DISCUSSION THREAD

**Discussion by elfren** [terjeio/grblHAL#259] [General]
Can grblHal handle stepper motor spindle?

**Comment by terjeio** [comment]
Code can be added at the driver level if the processor in question has timers available for step generation.

**Reply by elfren** [reply]
I would be using one of phil-barrett's Teensy 4.1 boards on a rose engine.  It's basically a lathe with a headstock which can move back and forth horizontally (controlled by cams called rosettes).  I can do some programming, but probably not good enough to do driver level stuff.

**Reply by terjeio** [reply]
Is it for just spinning the spindle at a programmed RPM or is motion to be synced with the spindle (or vice versa)?

There is some primitive code for moving a stepper outside of motion planner control in the Teensy driver, added for plasma torch height control. This could be used as a starting point if for RPM only control (no motion sync).

**Reply by elfren** [reply]
Ideally, it would need to sync with the X and Z axes.

**Reply by terjeio** [reply]
Which way? Can you use A-axis as a rotary axis for the spindle?

**Reply by elfren** [reply]
That could work. Currently we are using Z, X, and B, and no plans for an A.
